### PR TITLE
APPDEV-1130 Failed to retrieve details for user error

### PR DESCRIPTION
**Pull request by vishal-revdiwala**
- Handled the View stacks in both signup and edit profile scenes.
- Overridden back button visibility and action to support both flows.
- Tabs are hidden during Confirm OTP flow.
- Merged latest develop branch

### CODE REVIEW

File: Yona/Yona/Others/SMSValidPasscodeLogin/SMSValidation/ConfirmMobileValidationVC.swift [hunk 1]
```diff
@@ -71,19 +71,24 @@ extension ConfirmMobileValidationVC: CodeInputViewDelegate {
         Loader.Show()
         UserRequestManager.sharedInstance.confirmMobileNumber(body as BodyDataDictionary,onCompletion: { (success, message, serverCode )in
             Loader.Hide()
+            self.codeInputView.clear()
             if (success) {
                 self.codeInputView.resignFirstResponder()
                 //Update flag
                 if self.isFromUserProfile {
-                    self.navigationController?.popToRootViewController(animated: true)
+                    if UserDefaults.standard.bool(forKey: YonaConstants.nsUserDefaultsKeys.confirmPinFromProfile){
```
[the remark this answers is not in the document]
Comment (vishal-revdiwala) [reply Yona/Yona/Others/SMSValidPasscodeLogin/SMSValidation/ConfirmMobileValidationVC.swift]: Done.

File: Yona/Yona/SignUp/SignUpSecondStepViewController.swift [hunk 2]
```diff
@@ -52,26 +52,21 @@ class SignUpSecondStepViewController: BaseViewController,UIScrollViewDelegate {
...
     @IBOutlet var previousButton: UIButton!
-    
-    
     @IBOutlet weak var topViewHeightConstraint: NSLayoutConstraint!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         self.navigationController?.setNavigationBarHidden(false, animated: false)
         setupUI()
+        checkForUserHavePin()
```
[the remark this answers is not in the document]
Comment (vishal-revdiwala) [reply Yona/Yona/SignUp/SignUpSecondStepViewController.swift]: Updated.

File: Yona/Yona/SignUp/SignUpSecondStepViewController.swift [hunk 3]
```diff
@@ -251,25 +210,50 @@ class SignUpSecondStepViewController: BaseViewController,UIScrollViewDelegate {
...
     }
     
+    func userOverrideAction(formattedMobileNumber:String, body:Dictionary<String, String>) {
+        //alert the user ask if they want to override their account, if ok send back to SMS screen
+        let localizedString = NSLocalizedString("user-override", comment: "")
+        let title = NSString(format: localizedString as NSString, String(formattedMobileNumber))
+        self.displayAlertOption(title as String, cancelButton: true, alertDescription: "", onCompletion: { (buttonPressed) in
+            switch buttonPressed{
+            case alertButtonType.ok:
+                AdminRequestManager.sharedInstance.adminRequestOverride(body["mobileNumber"]) { (success, message, code) in
```
[the remark this answers is not in the document]
Comment (vishal-revdiwala) [reply Yona/Yona/SignUp/SignUpSecondStepViewController.swift]: Done.

File: Yona/Yona/SignUp/SignUpSecondStepViewController.swift [hunk 4]
```diff
@@ -251,25 +210,50 @@ class SignUpSecondStepViewController: BaseViewController,UIScrollViewDelegate {
         }
     }
     
+    func userOverrideAction(formattedMobileNumber:String, body:Dictionary<String, String>) {
```
[the remark this answers is not in the document]
Comment (vishal-revdiwala) [reply Yona/Yona/SignUp/SignUpSecondStepViewController.swift]: Done.

File: Yona/Yona/Others/SMSValidPasscodeLogin/SMSValidation/ConfirmMobileValidationVC.swift [hunk 5]
```diff
@@ -65,8 +65,23 @@ class ConfirmMobileValidationVC: ValidationMasterView {
 }
 
 extension ConfirmMobileValidationVC: CodeInputViewDelegate {
+    
+    func handleNavigtaionFromProfileView() {
```
Comment (Bert-R) [inline Yona/Yona/Others/SMSValidPasscodeLogin/SMSValidation/ConfirmMobileValidationVC.swift]: Typo: ``handleNavigtaionFromProfileView`` should be ``handleNavigationFromProfileView``.

Comment (vishal-revdiwala) [reply Yona/Yona/Others/SMSValidPasscodeLogin/SMSValidation/ConfirmMobileValidationVC.swift]: Updated.

**Pull request merged**